### PR TITLE
Refactor sdf::readXml

### PR DESCRIPTION
**Pull request by azeey**
## Summary
While working on a new feature for the `main` branch I got a codecheck error saying `sdf::readXml` is over 500 lines. This simply refactors two sections of the the function into helper functions.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

**Review by chapulina — APPROVED** [review]

File: src/parser.cc [hunk 1]
```diff
@@ -786,48 +786,14 @@ std::string getModelFilePath(const std::string &_modelDirPath)
 }
 
 //////////////////////////////////////////////////
-bool readXml(TiXmlElement *_xml, ElementPtr _sdf, Errors &_errors)
+/// Helper function read the all the attributes of an element from TinyXML to
```
Comment (aaronchongth) [inline src/parser.cc:789]:
```suggestion
/// Helper function that reads all the attributes of an element from TinyXML to 
```

File: src/parser.cc [hunk 2]
```diff
@@ -786,48 +786,14 @@ std::string getModelFilePath(const std::string &_modelDirPath)
...
+/// sdf::Element.
+/// \param[in] _xml Pointer to element to read the attributes from.
```
Comment (aaronchongth) [inline src/parser.cc:791]: Nit:
```suggestion
/// \param[in] _xml Pointer to XML element to read the attributes from.
```

File: src/parser.cc [hunk 3]
```diff
@@ -885,6 +851,115 @@ bool readXml(TiXmlElement *_xml, ElementPtr _sdf, Errors &_errors)
     }
   }
 
+  return true;
+}
+
+//////////////////////////////////////////////////
+/// Helper function to resolve file name from an //include/uri element.
+/// \param[in] _uriElem Pointer to the //include/uri element.
```
Comment (aaronchongth) [inline src/parser.cc:859]: Nit:
```suggestion
/// \param[in] _uriElem Pointer to the //include/uri XML element.
```

**Review by aaronchongth — APPROVED** [review]
Thanks for refactoring `readXml`! I think we were hitting the limit of allowed lines in a single function :laughing: I have a few trivial comments, but everything looks great!

File: src/parser.cc (hunk 1 above)
Comment (azeey) [reply src/parser.cc]: Thanks. https://github.com/ignitionrobotics/sdformat/pull/681/commits/6c1b2554955797539b0a287bb1fa85a51b998360

File: src/parser.cc (hunk 2 above)
Comment (azeey) [reply src/parser.cc]: https://github.com/ignitionrobotics/sdformat/pull/681/commits/6c1b2554955797539b0a287bb1fa85a51b998360

File: src/parser.cc (hunk 3 above)
Comment (azeey) [reply src/parser.cc]: https://github.com/ignitionrobotics/sdformat/pull/681/commits/6c1b2554955797539b0a287bb1fa85a51b998360

**Pull request merged**